.github/workflows/build.yaml

```diff
@@ -0,0 +1,49 @@
+name: Quick Go unit tests
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: off
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        path: ./src/github.com/${{ github.repository }}
+    - name: Cache go pkg directory
+      id: cache-go-pkg
+      uses: actions/cache@v1
+      with:
+        path: ${{ github.workspace }}/pkg
+        key: ${{ runner.os }}-go-pkg
+    - name: Cache go vendor packages
+      id: cache-go-vendor
+      uses: actions/cache@v1
+      with:
+        path: ${{ github.workspace }}/src/github.com/${{ github.repository }}/vendor
+        key: ${{ runner.os }}-go-vendor
+    # XXX: is this the right version of go?
+    - name: Set up go 1.10
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.10
+      id: go  # XXX: what is this for?
+    - name: Install govendor
+      run: go get -u github.com/kardianos/govendor
+    # This dominates the duration of the test. Can we get a smaller set that is
+    # enough to build embedded C parts?
+    - name: Get C dependencies
+      run: sudo apt-get build-dep -y ./src/github.com/${{ github.repository }}
+    - name: Get Go dependencies
+      run: |
+          cd ${{ github.workspace }}/src/github.com/${{ github.repository }}
+          ${{ github.workspace }}/bin/govendor sync
```

Comment:
How does the caching work? Is the above actions/cache@v1 enough to get this cached?

Reply:
Caching is an action that has pre/post code. The way I understand it we just save/restore a given directory.